rlog/2023-08-04-wakurtosis-waku-scallability.mdx

```diff
@@ -0,0 +1,174 @@
+In addition to this, several external modules developed in-house expand some of Kurtosis' limitations:
+
+- Network Generation Module (Gennet): Initiates and configures networks for the simulation. It's highly modular, supporting the integration of multiple topologies, protocols, and node traits.
+
+- Packet Injection Module (WLS): Allows for the insertion of custom data packets, thereby enabling varied traffic patterns and stress tests on the simulations.
+
+- Analysis Module: Captures and provides insights into resource usage, network behaviors, and protocol interactions throughout the enclave.
+
+### Data Collection  
+
+Wakurtosis ensures the accuracy of its data by leveraging multiple sources for hardware metrics:
+
+- ##### Cadvisor (a Google tool)
+  [Cadvisor](https://github.com/google/cadvisor) provides detailed metrics on resource usage and performance characteristics of Docker containers. Cadvisor monitors application containers at the individual level by directly interfacing with Docker's daemon API.
+While Cadvisor offers real-time metrics, it primarily focuses on container-specific metrics, which may neglect broader system-level insights.  
+
+- ##### Docker statistics
+  [Docker statistics](https://docs.docker.com/engine/reference/commandline/stats/) provides insights into Docker's overall performance and resource allocation.  
+This native Docker tool captures statistics about running containers using Docker's stats API, collecting cumulative CPU, memory, network, and block I/O metrics.
+Docker statistics offer a bird's-eye view of the system, which can sometimes miss the granularity of performance fluctuations inside individual containers, particularly when dealing with multiple processes per container.
+
+- ##### Process-level monitoring
+  Process-level monitoring offers detailed insights by tracking operational traits of processes within containers.
+This method employs deep inspection of individual processes running inside a container by accessing */proc* kernel files to gather real-time process statistics.  
+Reading from the */proc* filesystem offers a direct window into kernel statistics, providing comprehensive metrics on each process within the containers.
+However, while it offers granular insights, process-level monitoring can be more resource-intensive and might not always capture overarching system behavior.   
+
+### Performance Metrics
+
+Each simulation lasted 3 hours to reach a steady state.  
+
+- Hardware Level Metrics: Emphasis on memory usage, CPU consumption, disk I/O, and network I/O.
+
+- Topology Level Metrics: Focuses on the efficiency of message propagation across the network, including metrics like message injection, propagation times, and message loss.
+
+### Scalability  
+
+To overcome scalability challenges, Wakurtosis employs a multi-node approach, running several nodes within one container.
+This method supports simulations with over 1,000 nodes on a single machine.  
```

Comment:
The biggest network size mentioned below is 600 nodes. Why not using the full capacity of Wakurtosis?